public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -198,6 +199,22 @@ export class PanelQueryRunner {
     });
   }
 
+  cancelQuery() {
+    if (!this.subscription) {
+      return;
+    }
+
+    this.subscription.unsubscribe();
+
+    // If we have an old result with loading state, send it with done state
+    if (this.lastResult && this.lastResult.state === LoadingState.Loading) {
+      this.subject.next({
+        ...this.lastResult,
+        state: LoadingState.Done,
```

Comment:
not 100% convinced Done is the right status... but that is better than adding another status

Reply:
Can revisit after a PanelHeader UI make over (to show state icons more uniformly and extendable)